article/app/views/fragments/emailArticleBody.scala.html

```diff
@@ -128,6 +129,29 @@ <h2 class="block__title">@title</h2>
                         }
                     }
 
+                    case ContentAtomBlockElement(atomId) => {
+                        @page.article.content.media.map { atom: MediaAtom =>
+                            @if(atom.id == atomId) {
+                                @row(Seq("padded")) {
+                                    @EmailVideoImage.bestFor(atom.posterImage.get).map { imageUrl =>
+                                        @atom.assets.map { asset =>
+                                            @asset.platform match {
+                                                case MediaAssetPlatform.Youtube => {
+                                                    <a href="https://www.youtube.com/watch?v=@asset.id">
+                                                    @imgForArticle(imageUrl, Some(atom.title))
+                                                    </a>
+                                                }
+                                                case _ => {
+                                                    @imgForArticle(imageUrl, Some(atom.title))
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
```

Comment:
There's a lot of logic in the template that I'd ideally like to move to `BlockElement.scala`. However, `BlockElement.scala` has no knowledge of the atom data that comes back from CAPI (which is where I'm getting the placeholder image URL and the YouTube video ID).